astropy/coordinates/baseframe.py

```diff
@@ -236,17 +236,17 @@ class BaseCoordinateFrame(ShapedLikeNDArray, metaclass=FrameMeta):
 
     Parameters
     ----------
-    representation : `BaseRepresentation` or None
-        A representation object or `None` to have no data (or use the other
+    representation_type : `BaseRepresentation` or None
```

Comment:
I think this one should be changed to `data` - since here you are allowed to put in a `BaseRepresentation` *instance*.

Although I always found it annoying that in fact the initializer is `*args`. But that is independent of this PR?

Reply:
Yea, the `*args* thing annoys me too...I wonder if there is any way to use the metaclass to hack the signature? Sounds a little scary though. But yea, I think outside of the scope of this PR.